### PR TITLE
93 implement baton retry mechanism

### LINKED ISSUE #93: Implement Baton Retry Mechanism
Describe the solution you'd like**
Implement a retry mechanism for the **baton** subproject of the XPlane Plugin. Currently, baton attempts to connect once upon plugin start and then accepts defeat and panics.

### PR DESCRIPTION
**Pull request by hrmeiners**
### **What was changed?**

- For 90, fixed the build system failing our builds because of the xplane plugin file.
- For 93, added a retry connection mechanism in the baton subproject.

### **Why was it changed?**

- For 90, The C++ compiler for the GitHub Actions workflow "Meson Build System" requires this include as it uses a different version of the STL as the compiler that we run on our local machines. This compiler could be changed, but it would take too much effort when the C++ threading behavior that throws this error will be removed in this sprint.
- For 93, the mechanism was added so that no matter whether Relay or XPlane started first, the project would eventually connect.

### **How was it changed?**

- For 90, added the `#include <memory>` header in the xplane plugin file to deal with the Meson build errors that were happening.
- For 93, added the retry mechanism in the baton lib.rs file. I changed the single connection request into an infinite reconnect loop with an exponential backoff that starts at 100ms and caps off at 5 seconds. These numbers can be adjusted if desired.

I combined these two fixes into one PR as the CI fix was easy enough to fix.

### CODE REVIEW

File: xplane_plugin/subprojects/baton/src/lib.rs [hunk 1]
```diff
@@ -62,9 +63,24 @@ impl ThreadWrapper {
             // used for read/write operations
             let mut buffer = String::with_capacity(128);
 
+            // Connection retry loop with an exponential backoff capped at 5 seconds
+            let conn;
```
Comment (carlyrhoover) [inline xplane_plugin/subprojects/baton/src/lib.rs:67]: Would a maximum number of retries need to be implemented?

**Review by carlyrhoover — COMMENTED** [review]
Looks good, good job getting the retry done.

File: xplane_plugin/subprojects/baton/src/lib.rs (hunk 1 above)
Comment (hrmeiners) [reply xplane_plugin/subprojects/baton/src/lib.rs:67]: I like the idea, and the exponential backoff already makes baton eventually retry a connection once every 5 seconds, which works similar to this. I do agree that it can be useful, though, so I'll add this and pick a good sounding number!

Comment (hrmeiners) [reply xplane_plugin/subprojects/baton/src/lib.rs:67]: Actually, on second thought this does seem redundant with the back off mechanism in place. I'd rather the plugin keep trying to connect to Relay as long as it runs than just suddenly stop. @DeviPriya-Kolla do you agree?

**Comment by hrmeiners** [comment]
Converting to a draft so that I can do additional work on this PR with Relay.

**Comment by hrmeiners** [comment]
I force-rewrote the commit to this branch to the previous PR that was reviewed last week. I am running into too many confusing issues with modifying Relay to include a retry mechanism that it would serve me better to solve those in a separate branch: implement-relay-retry. This way, I can work on this more this week without making this PR languish in development hell.

File: xplane_plugin/subprojects/baton/src/lib.rs (hunk 1 above)
Comment (DeviPriya-Kolla) [reply xplane_plugin/subprojects/baton/src/lib.rs:67]: I see both sides. The backoff mechanism ensures retries don’t happen too frequently, but a max retry limit could help in some edge cases. My main concern is if there are situations where the plugin keeps retrying indefinitely even when a connection is unlikely to succeed (e.g., a misconfigured or unavailable Relay). Maybe we could log a warning after a certain number of retries to flag potential issues while still allowing the plugin to keep trying?

**Review by DeviPriya-Kolla — APPROVED** [review]
Great work on this implementation! Thank you for addressing issues #93 and #90 in this PR. One consideration for the future: while the current test coverage looks solid, adding more test cases—especially for edge scenarios—could help preempt potential blockers or unexpected errors down the line. This can be an area to expand in subsequent updates.

No blockers from my side. If no further comments from the team, this is good to merge.
